File: Old Java Code/Client/org/hardgate/familytree/registrations/BirthRegistration.java
/*
 * Created on 01-Jan-2005
 *
 */
package org.hardgate.familytree.registrations;

import java.io.Serializable;

import org.apache.log4j.Logger;
import org.hardgate.familytree.core.Fact;
import org.hardgate.familytree.core.ParentalGroup;

/**
 * @author db
 *
 */
public class BirthRegistration extends Registration implements Serializable {

	private static Logger logger = Logger.getLogger(BirthRegistration.class);
	private static final long serialVersionUID = 0;
	
	private Fact birth;

    public BirthRegistration (ParentalGroup familyGroup) {
        super(familyGroup);
        this.birth = familyGroup.getPreferredFact(Fact.BIRTH);
        registrationDate = (birth == null) ? null : birth.getFactDate();
    }

    public String getRegistrationLocation () {
        return getPlaceOfBirth();
    }

    public boolean isCertificatePresent() {
        return (birth == null) ? false : birth.isCertificatePresent();
    }
}

File: FTAnalyser/Registrations/BirthRegistration.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Xml;

namespace FTAnalyzer
{
    public class BirthRegistration : Registration {

	    private Fact birth;

        public BirthRegistration (ParentalGroup familyGroup) : base(familyGroup) {
            this.birth = familyGroup.getPreferredFact(Fact.BIRTH);
            registrationDate = (birth == null) ? null : birth.FactDate;
        }

        public override string RegistrationLocation
        {
            get { return PlaceOfBirth; }
        }

        public override bool isCertificatePresent()
        {
            return (birth == null) ? false : birth.isCertificatePresent();
        }
    }
}